processor/attributesprocessor/attribute_hasher.go

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package attributesprocessor
+
+import (
+	"crypto/sha1"
+	"encoding/binary"
+	"encoding/hex"
+	"math"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+const (
+	maxVarFloat64 = 8
+	maxVarInt64   = 8
+)
+
+// SHA1AttributeHahser hashes an AttributeValue using SHA1 and returns a
+// hashed version of the attribute. In practice, this would mostly be used
+// for string attributes but we support all types for completeness/correctness
+// and eliminate any surprises.
+func SHA1AttributeHahser(attr *tracepb.AttributeValue) *tracepb.AttributeValue {
+	var val []byte
+	switch attr.Value.(type) {
+	case *tracepb.AttributeValue_StringValue:
+		val = []byte(attr.GetStringValue().Value)
+	case *tracepb.AttributeValue_BoolValue:
+		if attr.GetBoolValue() {
+			val = []byte{1}
+		} else {
+			val = []byte{0}
+		}
+	case *tracepb.AttributeValue_IntValue:
+		val = make([]byte, maxVarInt64)
+		binary.PutVarint(val, attr.GetIntValue())
```

Comment:
Why Varint and not BigEndian like it is done for doubles?
Note that 0x7fffffffffffffff does not fit in 8 bytes for varint encoding so this will panic.

Also (nit), arguably LittleEndian may be a better choice given it is the dominant endiannes today.

Reply:
Done